src/citrine/resources/ingredient_run.py

```diff
@@ -119,3 +120,58 @@ class IngredientRunCollection(ObjectRunCollection[IngredientRun]):
     def get_type(cls) -> Type[IngredientRun]:
         """Return the resource type in the collection."""
         return IngredientRun
+
+    def list_by_spec(self, uid: Union[UUID, str], scope: str = 'id') -> Iterator[IngredientRun]:
+        """
+        [ALPHA] Get the ingredient runs using the specified ingredient spec.
+
+        Parameters
+        ----------
+        uid
+            The unique ID of the ingredient spec whose ingredient run usages are to be located.
+        scope
+            The scope of `uid`.
+        Returns
+        -------
+        Iterator[IngredientRun]
+            The ingredient runs using the specified ingredient spec.
+
+        """
+        return self._get_relation('ingredient-specs', uid=uid, scope=scope)
```

Comment:
Maybe use IngredientSpecCollection._collection_key (and similar elsewhere)?

Reply:
The collection key is slightly off because it uses an underscore. Nothing a `replace` couldn't fix but I'm not sure how much string manipulation we want to do for this. Thoughst?

Reply:
Is it worth trying to make use of the dataset agnostic path template, then?

Reply:
It would be awkward in this case because the dataset agnostic path you want would be from the relation's collection, yet the build method you want to use is in this collection. I don't see much utility to the dataset agnostic path (and other shared string templates) anyway...it eliminates a relatively low amount of copypasta and adds a lot more opacity to what the final rendered endpoint will actually look like.